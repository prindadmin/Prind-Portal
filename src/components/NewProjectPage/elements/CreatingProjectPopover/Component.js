import React, { Component } from 'react'
//import PropTypes from 'prop-types'

import PopOverHandler from '../../../common/popOverHandler'

import {
  Spinner,
  Intent,
} from '@blueprintjs/core'

import * as strings from '../../../../data/Strings'

export class Element extends Component {
  static propTypes = {
  }

  getContent = () => {
    return (
      <React.Fragment>
        {strings.PROJECT_CREATION_BILLING_WARNING}
        <Spinner size='150' intent={Intent.DANGER}/>
      </React.Fragment>
    )
  }

  render () {
    return (
      <PopOverHandler>
        <div id='popup-greyer' onClick={(e) => e.stopPropagation()}>
          <div id='project-creation-popover'>
            <div id='popup-box'>
              <div className='project-creation-popover-container' onClick={(e) => e.stopPropagation()}>
                <div className='element-title'>
                  {strings.PROJECT_CREATION_IN_PROGRESS}
                </div>
                <div className='element-description'>
                  {this.getContent()}
                </div>
              </div>
            </div>
          </div>
        </div>
      </PopOverHandler>
    )
  }

}

export default Element
