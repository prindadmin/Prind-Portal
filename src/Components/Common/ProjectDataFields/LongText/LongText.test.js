import React from 'react';
import { shallow } from 'enzyme';

import Component from './LongText'

it('Should render', () => {
  const props = {
    elementContent: {
      id: "2",
      title: "Long Text Box",
      description: "Long Text Box Description",
      editable: true,
      fieldDetails:{
        textValue: "Test save.",
      },
    },
    pageName: "occupation",
  };
  const component = shallow(<Component {...props} />);
  expect(component).toMatchSnapshot();
});
