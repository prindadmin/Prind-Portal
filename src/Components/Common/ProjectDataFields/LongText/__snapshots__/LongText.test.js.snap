// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render 1`] = `
<div
  id="long-text-element"
>
  <div
    className="long-text-element-container"
  >
    <div
      className="element-title"
    >
      Long Text Box
    </div>
    <div
      className="element-description"
    >
      Long Text Box Description
    </div>
    <div
      className="container"
    >
      <form
        className="form"
        onSubmit={[Function]}
      >
        <div
          className="row"
        >
          <div
            className="col"
          >
            <textarea
              disabled={false}
              id="textValue"
              name="textValue"
              onChange={[Function]}
              placeholder="Please provide extra details here..."
              value="Test save."
            />
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="col"
          >
            <input
              className="submit-button"
              id="submit"
              name="submit"
              readOnly={true}
              style={
                Object {
                  "maxWidth": "300px",
                }
              }
              type="submit"
              value="Save Changes"
            />
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;
